Readme script update (#1750)
analogdevicesinc/hdl: .github/scripts/README.md

> # Guidlines & files check scripts

Reviewer: ```suggestion
# Guidelines & files check scripts
```

> This scripts are used for checking the integrity and correctness of the
> files across HDL/ repository. 

Reviewer: ```suggestion
This script is used for checking the integrity and correctness of the README
files across the HDL repository. 
```

## User guide for [check_for_missing_readme_md.sh](https://github.com/analogdevicesinc/hdl/blob/main/.github/scripts/check_for_missing_readme_md.sh)

> 

Reviewer: ```suggestion
```

* It assumes a directory structure like:
  * `projects/<board>/README.md`.
  * `projects/<board>/<carrier>/README.md`.
* It requires `bash` or `sh`, but it does **not** require any external dependencies.

### Purpose

This script ensures that all HDL project directories contain properly formatted `README.md` files and that they follow documentation conventions.

---

### Rules that are checked

#### 1. Presence of README files

* Each board directory (e.g., `projects/adrv9009zu11eg/`) must contain a `README.md`.
* Each carrier subdirectory (e.g., `projects/adrv9009zu11eg/zcu102/`) must also contain a `README.md`.

#### 2. Title format

* The main board README must have a title in the format:  
  `<BOARD-NAME> HDL Project`  
  Example: `# ADRV9009ZU11EG HDL Project`

* Carrier README titles must follow the format:  
  `<BOARD-NAME>/<CARRIER-NAME> HDL Project`  
  Example: `ADRV9009ZU11EG/ZCU102 HDL Project`

* **Board names** are uppercased and underscores are replaced with hyphens (e.g., `ad9081_fmca_ebz` → `AD9081-FMCA-EBZ`).
* **Carrier names** are uppercased as-is (e.g., `de10nano` → `DE10NANO`), except for special carriers.

#### 3. Special carrier exceptions

> The following carriers are allowed to use hyphens in their README titles and are not flagged for title mismatches:

Reviewer: ```suggestion
The following carriers are allowed to use underscore in their README titles and are not flagged for title mismatches:
```
